### PR TITLE
Add missing docs and examples for fmt::Write

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -82,29 +85,79 @@ pub trait Write {
...
+    /// [`char`]: ../../std/primitive.char.html
+    /// [`Error`]: struct.Error.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fmt::{Error, Write};
+    ///
+    /// fn writer<W: Write>(f: &mut W, c: char) -> Result<(), Error> {
+    ///     f.write_char(s)
```
Comment (TimNN) [inline src/libcore/fmt/mod.rs]: This should be `f.write_char(c)`.

**Comment by frewsxcv** [comment]
r=me with the fixed error

File: src/libcore/fmt/mod.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libcore/fmt/mod.rs]: Indeed, thanks!

**Pull request merged**